Code/Mantid/Framework/DataObjects/src/ReflectometryTransform.cpp

```diff
@@ -223,8 +229,27 @@ Mantid::API::MatrixWorkspace_sptr ReflectometryTransform::execute(
   return ws;
 }
 
+boost::optional<boost::shared_ptr<Mantid::DataObjects::TableWorkspace>>
+createTable(boost::optional<
+            boost::shared_ptr<Mantid::DataObjects::TableWorkspace>> &vertexes) {
+  (*vertexes)->addColumn("double", "Qx"); //needs to be set to whatever our Transform is.
+  (*vertexes)->addColumn("double", "Qy"); //needs to be set to whatever our Transform is.
+  (*vertexes)->addColumn("int", "OriginIndex");
+  (*vertexes)->addColumn("int", "OriginBin");
+  (*vertexes)->addColumn("double", "CellSignal");
+  (*vertexes)->addColumn("double", "CellError");
+
+  return vertexes;
+}
+
 MatrixWorkspace_sptr ReflectometryTransform::executeNormPoly(
-    MatrixWorkspace_const_sptr inputWS) const {
+    MatrixWorkspace_const_sptr inputWS,
+    boost::optional<boost::shared_ptr<Mantid::DataObjects::TableWorkspace>>
+        &vertexes) const {
+  // Create a table for the output if we want to debug vertex positioning
+  if (vertexes) {
+    vertexes = createTable(vertexes);
```

Comment:
This call should happen closer to where the Table is first needed. Put it just before the outer for loop (the loop over the number of histograms)